rfcs/taskforce/0001-buildkiteci.md

```diff
@@ -0,0 +1,243 @@
+# BuildKite CI Switch
+
+===
+
+# Goal
+
+## Relevant Problems
+
+- CircleCI is expensive because it's hosted (P1)
+- Slow builds (P2)
+  - Our configuration is not scheduled optimally — this leads to slower builds than are actually possible (P2.1)
+    - Parallelism is not granular but rather just at the outskirts (P2.1.1)
+    - Test execution doesn't share artifacts before fanning out (P2.1.2)
+  - Instances are not as large as we'd want them to be (even at the "largest" size) (P2.2)
+  - Test execution is not granular (P2.3)
+- Requires a manual template expansion step (test.py) (P3)
+- Iterating on builds requires running things on our existing build infrastructure (P4)
+- Flaky tests don't have a good story ;; It isn't clear which tests are truly "flaky and we want to ignore the flake" and this failure is important to look at (P5)
+- Nix builds don't work in CircleCI's infrastructure anymore (P6)
+
+## Impact of this project
+
+Intended to solve or set up infrastructure for enabling solving all of the above
+friction points.
+
+# Design
+
+## Existing System
+
+We run the following sorts of jobs:
+
+1. Linting
+2. Non-consensus / Consensus compatibility
+3. Trace tool
+4. Update branch protection on GitHub (only on develop)
+5. Building the wallet + bots (macos)
+6. Building the website (macos)
+7. Archive nodes
+8. Build daemon (macos)
+9. Build client-sdk
+10. Artifact build
+11. Unit tests
+12. Nonconsensus unit tests
+13. Various integration tests
+14. Nightly builds with larger configurations that we don't pay attention to
+
+The longer builds are unit tests (30min !!!) and integration tests (~20min average ;; 8min is just the build). The artifact medium curves build (~45min !!!)
+
+### Breakdown of longest job (build-artifacts-medium-curves)
+
+1. Pinning opam packages ~2m
```

Comment:
We could eliminate the pinning, and require building a new container if pinned packages change, as in olden days.

Reply:
I'm hoping #4808 addresses this usage